### PR TITLE
Money Manager

### PR DESCRIPTION
**Pull request by NabilHY**
Thanks for offering your review on Capstone Project👍 :
- During the development of this project I have tried to meet the provided requirements : 
- ### Project requirements

#### Design
- Follow guidelines

#### Interactions
- Splash screen
  - A simple page with the name of the ap
- Sign up and login pages
  - The user should be able to register in the app with full name, email, and password 
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page, the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with the following:
    - name
    - amount 
    - categories
  - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use a device for authentication.
- You should validate all user input to ensure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:
image
 ## Video Presentation : 
> https://www.loom.com/share/4372afe4b19c40f1977637aefa117498
 ## Live Demo : 
> https://money-manag.herokuapp.com/

### CODE REVIEW

File: app/views/static_pages/splash_page.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            <%= link_to "Log out", destroy_user_session_path, data: { turbo_method: :delete }, class: 'btn btn-secondary'  %>
+          <% else %>
+          <div class="buttons">
+            <%= link_to "Log in", new_user_session_path, class: "btn btn-primary" %>
+            <%= link_to "Register", new_user_registration_path, class: "btn btn-secondary"  %>
+          </div>
+          <% end %>
+      </div>
+  </div>
+</div>
```
Comment (tresorsawasawa) [inline app/views/static_pages/splash_page.html.erb:17]: - [ ] Kindly make sure the authentication page sticks as much as possible on the template design and you should use the correct **font** on your project title text 

| Template design | Your Design |
| ------------------------- | ---------------------- |
| Screenshot from 2022-10-21 17-55-32 | Screenshot from 2022-10-21 18-02-30 |

File: app/assets/stylesheets/application.css.scss [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      margin: 0 auto;
+    }
+  }
+
+  .card {
+    width: 300px;
+    height: 300px;
+    color: #000;
+    background-color: #3979c3;
+    border-radius: 10px;
```
Comment (tresorsawasawa) [inline app/assets/stylesheets/application.css.scss:80]: **Notice**: here are the colors to be used for your design
Screenshot from 2022-10-21 18-13-13

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @NabilHY 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/static_pages/splash_page.html.erb (hunk 1 above)
Comment (DeliceLydia) [reply app/views/static_pages/splash_page.html.erb]: - [ ] This was not addressed

File: app/views/expenses/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                        <p class='display-4'><%= (expense.amount).to_i %>$</p>
+                    </div>
+                </div>
+            </div>
+        <% end %>
+    </div>
+</div>
+<div class="d-flex justify-content-center mt-5 mb-5">
+    <%= link_to "New Transaction", new_group_expense_path(@group.id), class: "btn btn-success" %>
+</div>
```
[the remark this answers is not in the document]
Comment (DeliceLydia) [reply app/views/expenses/index.html.erb]: Addressed

File: app/views/static_pages/splash_page.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (DeliceLydia) [reply app/views/static_pages/splash_page.html.erb]: addressed

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                <b>Created at :</b>
+                <p><%= (group.created_at).to_fs(:short) %></p>
+                </div>
+              </div>
+              <div>
+                <b>Amount :</b>
+                <p> <%= group.sum %> $</p>
+              </div>
+              <%= link_to "Transactions", group_expenses_path(group.id), class:"btn btn-primary btn-sm" %>
+            </div>
```
[the remark this answers is not in the document]
Comment (DeliceLydia) [reply app/views/groups/index.html.erb]: Resolved

File: app/views/groups/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+          </div>
+        </div>
+    <% end %>
+  </div>
+  <div class="d-flex justify-content-center mt-5 mb-5">
+    <%= link_to "New Category", new_group_path, class: "btn btn-success" %>
+  </div>
+</div>
```
[the remark this answers is not in the document]
Comment (DeliceLydia) [reply app/views/groups/index.html.erb]: resolved

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @NabilHY, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Required Changes ♻️
_Check the comments under the previous review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/static_pages/splash_page.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<div class="layouts">
+  <div class="parent">
+      <div class="card" id="leftToRight">
+          <h2>Money Manager</h2>
+          <div class="buttons">
```
Comment (codepantha) [inline app/views/static_pages/splash_page.html.erb:5]: - [ ] You have made a great effort in sticking to the template design just like the previous reviewer suggested. However, I still need to point out that you're not yet using the correct font family for the `h2` element. Please make this change 👍 
## Design Guideline
image
## Your Implementation
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @NabilHY 👋 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Good commit history

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/static_pages/splash_page.html.erb (hunk 6 above)
Comment (NabilHY) [reply app/views/static_pages/splash_page.html.erb:5]: I have already done done I updated the sass file to import the fertigo font and the h2 has a font family of it, maybe it's not working on your end

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Nabil, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**